### ISSUE TITLE
binary_sensor.speedtest_running does not update

### ISSUE THREAD

**Issue by jpettitt** [MrSuicideParrot/hassio-speedtest-addon#12]
When I launch speed test from an automation the sate of `binary_sensor.speedtest_running` does not change.  I'd like to be able to have an entity displayed for `binary_sensor.speedtest_running` with a tap action to start the test and have the entity state reflect that it's actually running but it doesn't seem to work.

My current workaround is to set a helper to "running" and and then wait for the upoload value to update and clear the helper.

```yaml
alias: Internet Speed Test
description: ""
trigger: []
condition: []
action:
  - service: input_boolean.turn_on
    data: {}
    target:
      entity_id: input_boolean.speed_test_running
  - service: hassio.addon_start
    data:
      addon: 6b87c29e_speedtest_addon
  - wait_for_trigger:
      - platform: state
        entity_id:
          - sensor.speedtest_upload
  - service: input_boolean.turn_off
    data: {}
    target:
      entity_id: input_boolean.speed_test_running
mode: single
```

**Comment by MrSuicideParrot** [comment]
I was able to replicate your issue. It's a little bit strange... I wasn't aware of the `binary_sensor.speedtest_running` sensor, but from my naive understanding of this feature, my addon wouldn't need to do anything to support it. I'll look into the code of homeassistant and try to understand what is happening.

**Comment by MrSuicideParrot** [comment]
This is more strange than I was expecting. Yesterday I enabled the `speedtest_running` sensor. I ran the addon manually and didn't change it from off to on. I didn't change anything, and on the last day changed its state twice. The addon is configured to run every 4 hours, so I don't understand why it registered only two times. I need to dig deeper... hahahahah

**Comment by jpettitt** [comment]
Sounds like a  [Heisenbug](https://en.wikipedia.org/wiki/Heisenbug)

**Comment by MrSuicideParrot** [comment]
From what I understood, this may be a bug in home assistant. It doesn't detect addons that run for a short amount of time.
I tested with the Let's Encrypt addon and obtained the same exact result. 
Additionally, I examined the code of the Home Assistant supervisor, hoping to find the responsible part for this, but I haven't been able to locate it yet.

Have you tested this sensor with other addons that run very quicky and are normally turned off?
